### PR TITLE
Chapter 4 6

### PR DESCRIPTION
**Pull request by virginiachu**
Uploaded cloudformation templates for chapter 4 and chapter 6

### CODE REVIEW

File: chapter-4/ch4-cloudwatchagent-ssm.yml [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+#  use this file except in compliance with the License. A copy of the        #
+#  License is located at                                                     #
+#                                                                            #
+#      http://aws.amazon.com/asl/                                            #
+#                                                                            #
+#  or in the "license" file accompanying this file. This file is distributed #
+#  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,        #
+#  express or implied. See the License for the specific language governing   #
+#  permissions and limitations under the License.                            #
+##############################################################################
```
Comment (bksarthak) [inline chapter-4/ch4-cloudwatchagent-ssm.yml]: Remove AWS license piece from the code files.

File: chapter-4/ch4-cloudwatchagent-ssm.yml [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+  InstanceAMI:
+    Description: Managed AMI ID for EC2 Instance
+    Type : String
+    Default: ami-098e42ae54c764c35
+  SSHLocation:
+    Description: The IP address range that can be used to SSH to the EC2 instances
+    Type: String
+    MinLength: '9'
+    MaxLength: '18'
+    Default: 52.95.4.9/32
```
Comment (bksarthak) [inline chapter-4/ch4-cloudwatchagent-ssm.yml]: no need for default here

File: chapter-4/ch4-cloudwatchagent-ssm.yml [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+    AllowedPattern: (\d{1,3})\.(\d{1,3})\.(\d{1,3})\.(\d{1,3})/(\d{1,2})
+    ConstraintDescription: must be a valid IP CIDR range of the form x.x.x.x/x.
+  SSMLogBucket:
+    Description: S3 Bucket
+    Type: String
+    Default: 578830841467-us-west-2-infra-12341234
```
Comment (bksarthak) [inline chapter-4/ch4-cloudwatchagent-ssm.yml]: remove default

File: chapter-4/ch4-cloudwatchlog-alarm.yml [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+#  use this file except in compliance with the License. A copy of the        #
+#  License is located at                                                     #
+#                                                                            #
+#      http://aws.amazon.com/asl/                                            #
+#                                                                            #
+#  or in the "license" file accompanying this file. This file is distributed #
+#  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,        #
+#  express or implied. See the License for the specific language governing   #
+#  permissions and limitations under the License.                            #
+##############################################################################
```
Comment (bksarthak) [inline chapter-4/ch4-cloudwatchlog-alarm.yml]: remove this AWS license block

File: chapter-4/ch4-cloudwatchlog-alarm.yml [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+    Description: 'cloudwatch log metric filter name space'
+    Default: "WebServer/400s"
+  LogFilterMetricName:
+    Type: String
+    Description: 'cloudwatch log metric filter name'
+    Default: "4xxCount"
+  SNSTopicForCWAlarms:
+    Type: String
+    Description: Provide SNS Topic ARN for standard CloudWatch alarms.
+    Default: arn:aws:sns:us-west-2:123456789012:SNS
```
Comment (bksarthak) [inline chapter-4/ch4-cloudwatchlog-alarm.yml]: remove default

File: chapter-4/ch4-cloudwatchlog-alarm.yml [hunk 6]
```diff
@@ -0,0 +1,204 @@
...
+    Description: 'cloudwatch anomaly time zone'
+    Default: "UTC"
+  AnomalyConfigExcludeStart:
+    Type: String
+    Description: 'cloudwatch anomaly exclude start time'
+    Default: "2022-07-01T00:00:00"
+  AnomalyConfigExcludeEnd:
+    Type: String
+    Description: 'cloudwatch anomaly exclude end time'
+    Default: "2022-07-01T23:59:59"
```
Comment (bksarthak) [inline chapter-4/ch4-cloudwatchlog-alarm.yml]: if exclude is same as include will it show the right data?

**Review by bksarthak — COMMENTED** [review]
there are some default values and the AWS license block which should be removed before merge

File: chapter-4/ch4-cloudwatchlog-alarm.yml (hunk 6 above)
Comment (spamneggs) [reply chapter-4/ch4-cloudwatchlog-alarm.yml]: The value is different to define a time range

**Pull request merged**